### PR TITLE
Lab message prefilling

### LINKED ISSUE #4106: Improve the mapping/prefilling the new case/new sample forms when processing a DEMIS LabMessage [2]
### Situation Description
Currently some of the fields that are needed to create a new case or new sample from a DEMIS Lab Message have to be entered manually. 
The expectation is that we read from the demis lab message and prefill the fields.
If no mapping is possible, we have to add it to a list, which can be discussed with HZI and Gematik.

Like:
DEMIS => Swab // SORMAS: Throat swab & Nasal swab
Lab => described // SORMAS: other lab = take over the information

### PR DESCRIPTION
**Pull request by vlad-ciucescu**
closes #4106 
related PR on Demis adapter side: https://git.vitasystems.dev/sormas/sormas-demis-adapter/-/merge_requests/20

### CODE REVIEW

File: sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java [hunk 1]
```diff
@@ -401,141 +412,181 @@ private void editPatogenTest(SampleDto sampleDto, PathogenTestDto testDto, int c
...
-			});
+			ControllerProvider.getSampleController().getSampleCreateComponent(sampleDto, () -> { });
 
 		CheckBox checkBox = sampleCreateComponent.getWrappedComponent().getField(Captions.sampleIncludeTestOnCreation);
 		checkBox.setValue(Boolean.TRUE);
 		checkBox.setEnabled(false);
 		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_RESULT)).setValue(labMessageDto.getTestResult());
 		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_TYPE)).setValue(labMessageDto.getTestType());
-		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TESTED_DISEASE)).setValue(labMessageDto.getTestedDisease());
+		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TESTED_DISEASE)).setValue(Disease.CORONAVIRUS);
```
Comment (FredrikSchaeferVitagroup) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java:536]: This is generic for the DEMIS adapter, not for labMessages in general. Please move setting such default values to the adapter.

File: sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java [hunk 2]
```diff
@@ -401,141 +412,181 @@ private void editPatogenTest(SampleDto sampleDto, PathogenTestDto testDto, int c
...
+		((NullableOptionGroup) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_RESULT_VERIFIED)).setValue(true);
```
Comment (FredrikSchaeferVitagroup) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java:537]: This is generic for the DEMIS adapter, not for labMessages in general. Please move setting such default values to the adapter.

**Review by FredrikSchaeferVitagroup — CHANGES_REQUESTED** [review]

File: sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java (hunk 2 above)
Comment (vlad-ciucescu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java:537]: I agree, but lab messages don't have a `verified` field. Should I add such a field or not prefill this field at all?

Comment (FredrikSchaeferVitagroup) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java:537]: I would prefer to have such a field (testResultVerified) added to the labMessageDto.

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 3]
```diff
@@ -6762,4 +6762,19 @@ ALTER TABLE facility ADD COLUMN areatype varchar(255);
...
+ALTER TABLE labmessage_history ADD COLUMN personemail VARCHAR(255);
+
+INSERT INTO schema_version (version_number, comment) VALUES (341, 'Add person email, person phone to LabMessage #4106');
+
+-- 2020-03-04 Add testResultVerified to LabMessage #4106
+ALTER TABLE labmessage ADD COLUMN testresultverified boolean DEFAULT false;
+ALTER TABLE labmessage_history ADD COLUMN testresultverified boolean DEFAULT false;
+UPDATE labmessage SET testresultverified = true;
+
+INSERT INTO schema_version (version_number, comment) VALUES (342, 'Add testResultVerified to LabMessage #4106');
```
Comment (FredrikSchaeferVitagroup) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:6778]: Wouldn't it be better to have just one new entry in the schema_version? The only ones required to modify databases are you and me.

**Review by FredrikSchaeferVitagroup — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/resources/sql/sormas_schema.sql (hunk 3 above)
Comment (vlad-ciucescu) [reply sormas-backend/src/main/resources/sql/sormas_schema.sql:6778]: Ok, I will concatenate them

File: sormas-backend/src/test/java/de/symeda/sormas/backend/sormastosormas/SormasToSormasLabMessageFacadeEjbTest.java [hunk 4]
```diff
@@ -111,6 +111,8 @@ private void setLabMessageFields(LabMessageDto labMessage, Date dateValue) {
 		labMessage.setTestType(PathogenTestType.CULTURE);
 		labMessage.setTestDateTime(dateValue);
 		labMessage.setTestResult(PathogenTestResultType.PENDING);
+		labMessage.setTestResultVerified(true);
+		labMessage.setTestResultVerified(true);
```
Comment (FredrikSchaeferVitagroup) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/sormastosormas/SormasToSormasLabMessageFacadeEjbTest.java:115]: twice?

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 5]
```diff
@@ -6762,11 +6762,22 @@ ALTER TABLE facility ADD COLUMN areatype varchar(255);
...
 
 INSERT INTO schema_version (version_number, comment) VALUES (341, 'Add event management status to Event #4255');
 
+-- 2021-03-04 Add person email, person phone, testResultVerified to LabMessage #4106
+ALTER TABLE labmessage ADD COLUMN personphone VARCHAR(255);
+ALTER TABLE labmessage ADD COLUMN personemail VARCHAR(255);
+ALTER TABLE labmessage_history ADD COLUMN personphone VARCHAR(255);
+ALTER TABLE labmessage_history ADD COLUMN personemail VARCHAR(255);
+ALTER TABLE labmessage ADD COLUMN testresultverified boolean DEFAULT false;
+ALTER TABLE labmessage_history ADD COLUMN testresultverified boolean DEFAULT false;
```
Comment (FredrikSchaeferVitagroup) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:6778]: Why are you setting the default constraint? I think SORMAS should not induce data by its own, here.

**Review by FredrikSchaeferVitagroup — CHANGES_REQUESTED** [review]

File: sormas-backend/src/test/java/de/symeda/sormas/backend/sormastosormas/SormasToSormasLabMessageFacadeEjbTest.java (hunk 4 above)
Comment (vlad-ciucescu) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/sormastosormas/SormasToSormasLabMessageFacadeEjbTest.java:115]: oops

**Review by FredrikSchaeferVitagroup — APPROVED** [review]

**Pull request merged**